docs/DeveloperGuide.md

```diff
@@ -1,34 +1,426 @@
+
+1. ResourceLoader - Component handles loading and saving of the task list and study area details to local storage.
+
+
+
+### 2.2. Task Component  
+  
+
+[comment]: # (@@author GanapathySanathBalaji)  
+  
+The Task component depends on 3 other components,   
+  
+ 1. Command Component - The Command component issues instructions for the Task component to execute based on the user's input.  
+ 
+ 2. UI Component - The UI component is used to display the results (and related exceptions) of the commands issued to the Task Component. 
+ 
+ 3. ResourceLoader Component - The ResourceLoader component is used to load the list of tasks stored previously when the application is started and is also used to store the current list of tasks to the local storage upon exit.   
+
+![Task Component](images/Task_Component_UML.png)   
+<div>Figure 2. Object Diagram for Task Component</div>
+<br>
+The task component contains 8 separate classes. They are as follows:  
+ 
+  1. Task : Abstract class used to model a generic task. 
+   
+  2. Event : Specialized task class used to model events.  
+  
+  3. Deadline : Specialized task class used to model a deadline.   
+   
+  4. TaskType : Enumeration class used to denote the various task types.   
+  
+  5. TaskList : Container class used to store list of tasks and handle related operations. 
+   
+  6. SchedulableTask : Class used to model a task which is scheduled based on user's requirements.  
+  
+  7. TaskComparator : Contains a custom comparator used to compare two schedulable tasks based on their 
+  numberOfDaysLeft attribute.  
+  
+  8.  TaskScheduler : Class used to check for feasibility and schedule a list of tasks based on user's requirements.     
+     
+
+[comment]: # (@@author NizarMohd)
+
+### 2.3. Study Area Component  
+   
+ The Study Area component depends on 3 other components,  
+   
+  1. Command Component - The Command component issues instructions for the Study Area component to execute based on the 
+  user's input. 
+  
+  2. UI Component - The UI component is used to display the results (and related exceptions) of the commands issued to 
+  the Study Area Component.
+  
+  3. ResourceLoader Component - The ResourceLoader component is used to load data of all existing Study Areas in NUS 
+  into text files when the User first run the software. Eventually, data will be referred from the created text file.   
+  
+![Study Area Component](images/StudyAreaObjectDiagram.png) 
+<div>Figure 3. Object diagram for Study Area Component</div>
+<br>
+ The Study Area component contains 3 separate classes. They are as follows:  
+  
+   1. Dictionary : Class to map user input to specific terms used in StudyAreaList.
+    
+   2. StudyArea : Class that is used to model Study Areas. 
+   
+   3. StudyAreaList : Class that handles the list of available Study Areas based on User input.  
+
+
+[comment]: # (@@author terrytay)
+
+### 2.4. Notes Component
+![Notes Component](images/NotesComponent.png)
```

Comment:
I felt the arrow leading from ModuleManager was quite confusing as the branch suddenly became curved, and almost overlaps the top arrow. This might make some look for a separate origin rather than thinking of a split. 

It might be helpful to describe what CommandStack is for as it's the only floating unlinked object.